include/fs_utils.h

```diff
@@ -32,6 +32,37 @@
 
 #include "std_filesystem.h"
 
+#if defined(WIN32)
+#include <windows.h>
+using NativeFileHandle = HANDLE;
+// Cannot be constexpr due to Win32 macro
+#define InvalidNativeFileHandle INVALID_HANDLE_VALUE
+#else
+using NativeFileHandle = int;
+constexpr NativeFileHandle InvalidNativeFileHandle = -1;
+#endif
```

Comment:
Maybe format it like this to make it look marginally nicer:

```cpp
#if defined(WIN32)
	#include <windows.h>

	using NativeFileHandle = HANDLE;
	// Cannot be constexpr due to Win32 macro
	#define InvalidNativeFileHandle INVALID_HANDLE_VALUE

#else // Linux, macOS
	using NativeFileHandle = int;
	constexpr NativeFileHandle InvalidNativeFileHandle = -1;
#endif
```

Reply:
fixed